paddle/operators/cross_entropy_op.cu

```diff
@@ -12,10 +12,123 @@
    See the License for the specific language governing permissions and
    limitations under the License. */
 
-#define EIGEN_USE_GPU
-#include "paddle/operators/cross_entropy_op.h"
+#include "paddle/framework/op_registry.h"
+#include "paddle/platform/assert.h"
+
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+
+template <typename T>
+struct clipping_log {
```

Comment:
I notice that Majel and WarpCTC use many [`functors`](https://stackoverflow.com/questions/356950/c-functors-and-their-uses) to define many common functions. The functor has its own advantages, see:  https://stackoverflow.com/questions/356950/c-functors-and-their-uses

But it is not necessary here, I change it to the plain function.